diff.py

```diff
@@ -1880,21 +1950,136 @@ def progress(self, msg: str) -> None:
         sys.stdout.write("\x1b7\x1b[1;1f{}\x1b8".format(msg + " "))
         sys.stdout.flush()
 
-    def update(self, text: str, error: bool) -> None:
-        if not error and not self.emsg and text == self.mydump:
+    def update(self, mydump: str) -> None:
+        if not self.emsg and mydump == self.mydump:
             self.progress("Unchanged. ")
             return
-        self.pending_update = (text, error)
-        if not self.less_proc:
+        self.pending_update = (mydump, False)
+        self.kill_less()
+
+    def update_error(self, error: str) -> None:
+        self.pending_update = (error, True)
+        self.kill_less()
+
+    def terminate(self) -> None:
+        self.pending_update = None
+        self.kill_less()
+
+
+class WebDisplay(Display):
+    watch_queue: "queue.Queue[Optional[float]]"
+    ready_queue: "queue.Queue[Optional[float]]"
+    http_server: http.server.HTTPServer
+    running: bool
+
+    def run_sync(self) -> None:
+        web_display = self
+
+        class MyReqHandler(http.server.BaseHTTPRequestHandler):
+            def do_GET(self):
+                if not web_display.running:
+                    return
+                self.send_response(http.HTTPStatus.OK)
+                self.send_header("Access-Control-Allow-Origin", "*")
```

Comment:
(seems sketchy)

Reply:
`Access-Control-Allow-Origin: *` ? The header is needed for the browser to allow javascript to make the request

I could use `null` instead of `*`, at least on firefox, not sure if it's standard for file:// origins to have `Origin: null`

I could also use JSONP (the web technology has incredibly bad names) to not deal with CORS but that's way more sketchy imo